TYPO3.Neos/Resources/Public/JavaScript/Content/Inspector/Editors/AssetEditor.js

```diff
@@ -141,7 +142,9 @@ function(Ember, $, FileUpload, template, SecondaryInspectorController, Utility,
 
 			if (assetIdentifiers.length > 0) {
 				this.set('_showLoadingIndicator', true);
-				HttpClient.getResource(that.get('_assetMetadataEndpointUri') + '?' + $.param({assets: assetIdentifiers})).then(
+				// overriding the request method to POST, so too many assets do not cause "Request URI Too Large" errors.
+				// settings CSRF token manually, since getResource (naturally) does not add it.
+				HttpClient.getResource(that.get('_assetMetadataEndpointUri'), {type: 'POST', data: {__csrfToken: csrfToken, assets: assetIdentifiers}}).then(
```

Comment:
hmm `HttpClient` doesn't really allow for this, although pretty normal.. there's a few other ways to do this.

One way would be to alter `getRequest` to accept an optional requestMethod argument, alternatively just use `createResource` instead of `getRequest`
At the very least the CSRF token should not be read from DOM directly, use the `Configuration` object like it's done in `HttpClient` itself

Reply:
Mh, maybe it _is_ worth iterating through `assetIdentifiers` in batches of X and keep the `GET` requests..

Reply:
@aertmann what do you mean with
> hmm `HttpClient` doesn't really allow for this

What "this" is it you are referring to? And I definitely don't want to use `createResource()`, since that is just semantically wrong.

Will fix fetching the CSRF token…

Reply:
fetching something using a POST request, it implies POST requests only work for creating resources..

so you'd have to use createResource for POST requests which sounds wrong, but that's just due to the naming used for the methods..

and overriding type and handling CRSF tokens outside the HttpClient seems wrong as well, but whatever not that big of a deal

Reply:
It allows for it just fine (hehe) using the `type` override. And it feels better to do that, than to call `createResource()`.

What actually is more of a blocker: if a `GET` or `HEAD` request is done, jQuery forces the use of the URI, even if you pass data for the body. Because just sending the data in the body works just fine, and nowhere does any spec prohibit sending a body with a `GET` (or `HEAD`).

Anyway… this way it works fine. :)